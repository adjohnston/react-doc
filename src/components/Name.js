import React from 'react'
import PropTypes from 'prop-types'
import classString from 'src/helpers/class-string'

const FallbackTemplate = (name) => (!!name && (
  <h1
    className={`${classString('__name')}`}>
    {name}
  </h1>
)) || null

const Name = ({children, name}) => (
  (children && name)
    ? children(name)
    : FallbackTemplate(name)
)

Name.propTypes = {
  children: PropTypes.func,
  name: PropTypes.string
}

FallbackTemplate.propTypes = {
  name: PropTypes.string.isRequired
}

export default Name
