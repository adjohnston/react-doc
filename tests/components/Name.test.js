import React from 'react'
import renderer from 'react-test-renderer'
import Name from 'src/components/Name'

test('it should render correctly with `name` as a string', () => {
  const mockProps = {
    name: 'Component Name'
  }

  const tree = renderer
    .create(<Name {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly when given a template', () => {
  const mockProps = {
    name: 'A test component',
    children: (name) => (
      <h1
        className='custom-template'>
        Component: {name}
      </h1>
    )
  }

  const tree = renderer
    .create(<Name {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render null with no children', () => {
  const tree = renderer
    .create(<Name />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
