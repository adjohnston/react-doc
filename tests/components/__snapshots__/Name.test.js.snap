// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly when given a template 1`] = `
<h1
  className="custom-template"
>
  Component: 
  A test component
</h1>
`;

exports[`it should render correctly with \`name\` as a string 1`] = `
<h1
  className="stncl__name"
>
  Component Name
</h1>
`;

exports[`it should render null with no children 1`] = `null`;
